pkg/operator/ceph/cluster/mgr/spec.go

```diff
@@ -318,10 +318,10 @@ func (c *Cluster) makeDashboardService(name, activeDaemon string) (*v1.Service,
 	return svc, nil
 }
 
-func (c *Cluster) getPodLabels(daemonName string, includeNewLabels bool) map[string]string {
-	labels := controller.CephDaemonAppLabels(AppName, c.clusterInfo.Namespace, "mgr", daemonName, includeNewLabels)
+func (c *Cluster) getPodLabels(mgrConfig *mgrConfig, includeNewLabels bool) map[string]string {
+	labels := controller.CephDaemonAppLabels(AppName, c.clusterInfo.Namespace, config.MgrType, mgrConfig.DaemonID, "rook-operator", "cephclusters.ceph.rook.io", includeNewLabels)
```

Comment:
This should give the parent name. We really shouldn't need to hard-code "rook-operator" anywhere in the code, except one other place commented later for discovery.
```suggestion
	labels := controller.CephDaemonAppLabels(AppName, c.clusterInfo.Namespace, config.MgrType, mgrConfig.DaemonID, c.clusterInfo.NamespacedName(), "cephclusters.ceph.rook.io", includeNewLabels)
```

Reply:
Actually, I am getting panic from using this.
Check this particular line. https://github.com/rook/rook/blob/c3b5d3b078f096e6eaa5d0b5d0fcbe1206b0b071/pkg/daemon/ceph/client/info.go#L84

Reply:
Is the panic from running unit tests, integration tests, or from a manual test? If unit tests we will need to set the cluster name property in the tests. If in the real scenarios, we should look for a fix.

Reply:
Yes, it's a real scenario, the operator goes to crashloopbackoff state.

Reply:
Yes found a bug, created a new issue https://github.com/rook/rook/issues/9159